### ISSUE TITLE
Handling of \l

### ISSUE THREAD

**Issue by kirbyfan64** [kevinlawler/kona#353]
In k2:

```
ryan@DevPC-LX:~/kc$ rlwrap k2
K 2.8 2000-10-10 Copyright (C) 1993-2000 Kx Systems
Evaluation. Not for commercial use. 
\ for help. \\ to exit.

  \l kc
  / no more output

```

Kona:

```
K Console - Enter \ for help

  \l kc
"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
"0123456789"
"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"
{~(#x)=x?y}
{x@y}
{~(#x)=x?y}["ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";]
...
{1++/{#&x="\n"}'x#y[;1]}
{((`error;x;calcp[z;y]);-1+#y)}
{:[y=*x;z x;parsex["expected `",($y),", got `",$*x]]}

  / lots of output...

```

**Comment by tavmem** [comment]
Kona can get quite verbose.
This was an early, intentional deviation from k.
Cutting down on the amount of output could certainly be viewed as an enhancement.

**Comment by kirbyfan64** [comment]
I don't really mind either way. I was just curious on whether or not it was intentional.

**Comment by tavmem** [comment]
Most scripts use amend to create and save functions and/or variables in the workspace.
The verbose behavior of Kona with scripts is just a generalized case of Kona displaying output after amend is executed.  In k2.8
```
  v:3
  f:{x}
```
In Kona
```
  v:3
3
  f:{x}
{x}
```

**Comment by tavmem** [comment]
But ... the amend has to be the last operation done in the command.  In k2.8
```
  v:3
  f:{x}
  3+v:4
7
```

**Comment by kirbyfan64** [comment]
Yeah, I actually prefer Kona's behavior at this point. It can make debugging scripts somewhat easier.

**Comment by tavmem** [comment]
There is also the multiple-commands-on-a-line case to consider.  In k2.8
```
  3+v:4
7
  3+v:4; v:9
  3+v:4; 2+v:9
11
```

**Comment by tavmem** [comment]
This modification brings Kona in line with the family of similar languages:
APL, A+, J, K, and Q
Try it.  We can always put it back the other way.

**Comment by effbiae** [comment]
scripts could be modified so each line ends in ; like
```
  \cat f.k
x:!10;
+/x;
  \l f.k  
  x
0 1 2 3 4 5 6 7 8 9
```
but that's unkish. maybe introduce an output mode with \o so:
```
  \o 0
  !10 
  \o 1
  !10
0 1 2 3 4 5 6 7 8 9
  \o 
1
```
or maybe provide a redirect for \f like \f f.k /dev/null 

i'm happy to provide a patch if it's worth it.

**Comment by effbiae** [comment]
i approve the amend print suppression by @tavmem https://github.com/kevinlawler/kona/commit/43b5962366f215038d012cdaf665c44fcafec01c but as stated earlier, kona made this deviation from k consciously and the commit probably won't make it in.

**Comment by tavmem** [comment]
This was fixed by commit 0e6351523ba28def719fe44f1c994764ea57ac02
